apps/web/src/components/Button/ButtonWithLink.tsx

```diff
@@ -0,0 +1,58 @@
+"use client"
+
+import Link, { LinkProps } from 'apps/web/node_modules/next/link';
+
+import { Button, ButtonProps, ButtonVariants, ButtonSizes } from './Button';
+
+export function ButtonWithLink({
+  href,
+  target,
+  rel,
+  linkClassNames,
+  linkProps,
+  variant = 'Primary',
+  size = 'Medium',
```

Comment:
I don't think you need to set variant / size, since button already defaults to primary / medium.

Reply:
addressed by retyping my implementation to use `ButtonVariants` and `ButtonSizes` types instead of strings.